### PR TITLE
https://trello.com/c/I6eDHyd5/10503-cursor-issue-arabic

### PR DESCRIPTION
**Pull request by brawaga**
JS target solution only. Non-breaking space no more LTR character in Haxe TextClip natives, alphabet direction function added, improved metrics for Arabic word parts via fooling pixijs.

Linked PR: https://github.com/area9innovation/innovation/pull/1808.

### CODE REVIEW

**Comment by brawaga** [comment]
4 tests in innovation and 2 tests in Rhapsode fail.

File: platforms/js/TextClip.hx [hunk 1]
```diff
@@ -403,8 +403,10 @@ class TextClip extends NativeWidgetClip {
 
 		advanceCorrection = untyped (getAdvancedWidthsCorrection(tm, style, eochi, egchi, egb)-getAdvancedWidthsCorrection(tm, style, bochi, bgchi, bgb)) / UPM * style.fontSize;
```
Comment (ykuzmenko73) [inline platforms/js/TextClip.hx:405]: Please, add comments about the idea

Comment (brawaga) [reply platforms/js/TextClip.hx:405]: This was here for ages, but ok, I will remember what this was about.

Comment (brawaga) [reply platforms/js/TextClip.hx:405]: UPD: this was done for same goal scriptingFixSuffix invented for, but for some reason did not work recently. I will remove it and see if working.

**Pull request merged**